stl/inc/xstring

```diff
@@ -2630,6 +2640,20 @@ private:
         _Al.deallocate(_Old_ptr, _Capacity + 1); // +1 for null terminator
     }
 
+    _CONSTEXPR20 void _Construct_empty() noexcept {
+        // this function needs to be noexcept for conformance to the standard.

```

Comment:
I'm not too sure whether it's good to add `noexcept` for this function.

Reply:
It doesn't *need* to be noexcept - any (non-conformant) exceptions thrown by the proxy allocation will slam into the `basic_string` constructor's annotation. Arguably, this shouldn't actually be marked - note that `basic_string() noexcept(is_nothrow_default_constructible_v<_Alty>)` permits exceptions to be emitted (although this is from a surprising source). I think we should remove this marking and the comment.

Reply:
It's also a behavioral change compared to the previous direct allocation of a proxy.